cmd/collector/app/thrift_span_handler.go

```diff
@@ -35,22 +34,21 @@ const (
 	UnknownFormatType = "unknown"
 )
 
+// SubmitBatchOptions are passed to Submit methods of the handlers.
+type SubmitBatchOptions struct {
+	InboundTransport string
```

Comment:
Same as above: what are the implications of having the `InboundTransport` not being specified?

Reply:
I have a followup story to take care of that. I think if `InBoundTransport` is not set, we would simply skip setting another dimension for span counter metrics.

Reply:
We shouldn't skip, but assign an "undefined" value so that the total counter is still correct.

Reply:
> We shouldn't skip, but assign an "undefined" value so that the total counter is still correct.

hmmm. What I meant was something like the following:
```
    tags := map[string]string{"svc": serviceName, "debug": debugStr}
    if endpointType != "" {
	  tags["transport"] = endpointType
    }
    c := m.factory.Counter(metrics.Options{Name: m.category, Tags: tags})
```
so if endpointType is empty we still count it just not having a transport tag. Is adding `undefined` tag mandatory here for metrics to work correctly?

Reply:
Well, first, we don't create counter objects with labels in real time, because it's expensive and we always prefer to pre-create them when possible and cache. Second, when creating a counter we must define a set of labels (tags) it has. Some metrics backends may not like the value of the label to be blank. Why not be explicit about the name then?
```
    tags := map[string]string{"svc": serviceName, "debug": debugStr,  "transport": "undefined"}
    if endpointType != "" {
	  tags["transport"] = endpointType
    }
```

Reply:
gotcha